kolibri/utils/server.py

```diff
@@ -289,11 +289,22 @@ def STOP(self):
 class ZeroConfPlugin(Monitor):
     def __init__(self, bus, port):
         self.port = port
-        Monitor.__init__(self, bus, self.run, frequency=5)
+        if conf.OPTIONS["Deployment"]["LISTEN_ADDRESS"] == "0.0.0.0":
+            # Only bother doing dynamic updates of the zeroconf service if we're bound
```

Comment:
why does listening on a single one of the device interfaces blocks dynamic updates?

Reply:
The dynamic updates are only required when new IP addresses are available to broadcast/listen on in zeroconf. If we have bound to a specific listen address, then this is not necessary, as we are always listening/broadcasting only on that specific listen address.

If we've bound to 0.0.0.0 then if a new IP address is now bound to a local interface, we should start listening and broadcasting on it.

Reply:
thanks for the explanation!